docs/layer_configuration.md

```diff
@@ -0,0 +1,342 @@
+
+Vulkan layers give Vulkan application developers the ability to add additional functionality to applications without modifying the application itself, e.g., validating API usages, dumping API entry points or generating screenshots of specified frames.
+
+Vulkan layers can be configured using three different methods to match specific Vulkan developers workflows:
+- Using the Vulkan API: `VkCreateInstance()` and `VK_EXT_layer_settings`.
+- Using the `vk_layer_settings.txt` file, that can be generated by the GUI interface called *[Vulkan Configurator](https://vulkan.lunarg.com/doc/sdk/latest/windows/vkconfig.html)*.
+- Using environment variables.
+
+These three methods are implemented by the **Vulkan Layer Settings** library part of the *[Vulkan-Utility-Libraries](https://github.com/KhronosGroup/Vulkan-Utility-Libraries)* repository.
+Any layer project including this library will provide these three methods to control layer settings bringing consistency and easy of use of layers to the Vulkan community.
+
+Configuring layers means multiple tasks: Enabling layers; Ordering layers; Configuring the layers capabilities. These three aspects are described with each method to configure layers.
+
+A single setting can be configure using all three methods simulanously. A convention defines priorities regarding the different layer settings methods:
+1. The environment variable which overrides
+2. The vk_layer_settings.txt value which overrides
+3. The `VK_EXT_layer_settings` value of a layer setting.
+
+**Guideline: Settings which are unknown by the layer will be ignored independently of the method. It's the reponsability of the layer developer to ensure backward compatibility with previous versions of the layer. This is to ensure the list of layer settings remain stable across versions and that the responsability of handling layer backward compatibility doesn't fall on Vulkan application developers as this could quickly become untrackable.**.
+
+## Configuring Vulkan Layers using the *Vulkan API*
+
+### Enabling and ordering the layer using `vkCreateInstance()`
+
+Applications may programmatically activate layers via the `vkCreateInstance()` entry point. This
+is done by setting `enabledLayerCount` and `ppEnabledLayerNames` in the `VkInstanceCreateInfo`
+structure.
+
+The layer names order in `ppEnabledLayerNames` specifies the layers execution ordering from closer to the Vulkan application to closer
+to the Vulkan driver.
+
+### Configuring the layer settings using `VK_EXT_layer_settings`
+
+Layer settings may be configured using the `VK_EXT_layer_settings` extension by initializing the `VkLayerSettingsCreateInfoEXT` structure and chaining it to the `pNext` of `VkInstanceCreateInfo` when creating a Vulkan instance.
+
+### Code example
```

Comment:
To keep the example simple I'd just use the validation layers. Using the validation layers and profiles complicates the example making it harder to follow.

If you feel it valuable to show how to set multiple layers I'd make another example.

Reply:
Ok, I updated all the examples to use the validation layer, and actually configuring it the same way with the different method.

I also used the profiles layer to express the layer ordering in the exemple.